python/lance/_lib.pyx

```diff
@@ -311,3 +306,116 @@ cdef class FileSystemDataset(Dataset):
 def _wrap_dataset(Dataset dataset not None):
     cdef shared_ptr[CDataset] copy = dataset.unwrap()
     return FileSystemDataset.wrap(move(copy))
+
+
+cdef extern from "lance/arrow/dataset.h" namespace "lance::arrow" nogil:
+    cdef cppclass CLanceDataset "::lance::arrow::LanceDataset":
+        enum WriteMode "WriteMode":
+            CREATE "::lance::arrow::LanceDataset::WriteMode::kCreate"
+            APPEND "::lance::arrow::LanceDataset::WriteMode::kAppend"
+            OVERWRITE "::lance::arrow::LanceDataset::WriteMode::kOverwrite"
+
+        @ staticmethod
+        CStatus Write(
+                const CFileSystemDatasetWriteOptions& write_options,
+                shared_ptr[CDataset] dataset,
+                WriteMode mode)
+
+        @ staticmethod
+        CResult[shared_ptr[CLanceDataset]] Make(
+                const shared_ptr[CFileSystem]& fs,
+                const string& base_uri,
+                optional[uint64_t] version,
+        )
+
+
+
+cdef class LanceDataset(Dataset):
```

Comment:
For DuckDB compat (current release), this needs to be called `FileSystemDataset` since they're doing a string name check